dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -651,9 +651,16 @@ void MergeTreeData::clearOldTemporaryDirectories(ssize_t custom_directories_life
                     Poco::File(full_path + it.name()).remove(true);
                 }
             }
-            catch (const Poco::FileNotFoundException &)
+            catch (const Poco::FileNotFoundException & ex)
             {
                 /// If the file is already deleted, do nothing.
+                LOG_WARNING(log, ex.what() << " (concurrent directory removing, ignored)");
+            }
+            catch (const Poco::DirectoryNotEmptyException & ex)
+            {
+                // MergeTreeDataMergerMutator is merging the partition, in NFS, we will get DirectoryNotEmptyExcetion
```

Comment:
Is the problem specific to NFS? I guess if the merger is concurrently writing into the `tmp_merge_` directory, the removing process can easily miss the column files that were just added and this problem will manifest even on the ordinary FS.

Reply:
Still I am wary to catch very general exceptions and blame it on NFS. This can lead to misleading log messages - suppose a user who doesn't use NFS gets this error message. That's confusing.

I think this error is just more likely with NFS, but can happen even on ordinary FS - if you have one process that adds files (a merge process) and another that deletes files (clearOldTemporaryDirectories) - DirectorNotEmptyExceptions can happen.